sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/RateLimitPolicyFactory.java

```diff
@@ -67,4 +84,55 @@ public void onSuccess(List<KinesisRecord> records) throws InterruptedException {
       Thread.sleep(delay.get().getMillis());
     }
   }
+
+  /**
+   * Default rate limiter that throttles reading from a shard using an exponential backoff if the
+   * response is empty or if the consumer is throttled by AWS.
+   */
+  class DefaultRateLimiter implements RateLimitPolicy {
```

Comment:
Ii would be great to add a test for this class.

Reply:
Done ✅